### PR TITLE
Convert redeploy.py to buildkite

### PR DESCRIPTION
**Pull request by dbkr**
Hopefully fairly self-explanatory

### CODE REVIEW

File: scripts/redeploy.py [hunk 1]
```diff
@@ -204,13 +226,33 @@ def deploy_tarball(tar_gz_url, build_dir):
...
+        "--api-token", dest="api_token", help=(
+            "API access token for buildkite. Require read_artifacts scope."
+        ), required=True,
+    )
+
+    # We require a matching webhook token, but because we take everything else
+    # about what to deploy from the poke body, we can be a little more paranoid
+    # and only accept builds / artifacts from a specific buildkite org
+    parser.add_argument(
+        "--org", dest="buildkit_org", help=(
```
Comment (jryans) [inline scripts/redeploy.py]:
```suggestion
        "--org", dest="buildkite_org", help=(
```
here and below

File: scripts/redeploy.py [hunk 2]
```diff
@@ -16,17 +16,20 @@
...
 
 from deploy import Deployer, DeployException
 
 app = Flask(__name__)
 
-arg_jenkins_url = None
 deployer = None
 arg_extract_path = None
 arg_symlink = None
+arg_webbook_token = None
```
Comment (jryans) [inline scripts/redeploy.py]:
```suggestion
arg_webhook_token = None
```

File: scripts/redeploy.py [hunk 3]
```diff
@@ -2,7 +2,7 @@
 #
 # auto-deploy script for https://riot.im/develop
 #
-# Listens for HTTP hits. When it gets one, downloads the artifact from jenkins
+# Listens for HTTP hits. When it gets one, downloads the artifact from buildkite
```
Comment (jryans) [inline scripts/redeploy.py]: Maybe link to  https://buildkite.com/docs/apis/webhooks somewhere in the file.

File: scripts/redeploy.py [hunk 4]
```diff
@@ -39,81 +42,98 @@ def create_symlink(source, linkname):
...
+    if event is None:
+        abort(400, "No 'event' specified")
         return
 
-    build_num = incoming_json.get("build", {}).get("number", 0)
-    if not build_num or build_num <= 0 or not isinstance(build_num, int):
-        abort(400, "Missing or bad build number")
+    if event == 'ping':
+        print("Got ping request - responding")
+        return jsonify({'response': 'pong!'})
```
Comment (jryans) [inline scripts/redeploy.py]: I guess the response can be anything here?

Comment (dbkr) [reply scripts/redeploy.py]: yep

File: scripts/redeploy.py (hunk 3 above)
Comment (dbkr) [reply scripts/redeploy.py]: done

**Pull request merged**